src/webui/webapplication.cpp

```diff
@@ -318,7 +342,13 @@ void WebApplication::action_command_download()
             if ((url.size() == 40 && !url.contains(QRegExp("[^0-9A-Fa-f]")))
                 || (url.size() == 32 && !url.contains(QRegExp("[^2-7A-Za-z]"))))
                 url = "magnet:?xt=urn:btih:" + url;
-            BitTorrent::Session::instance()->addTorrent(url);
+
+            BitTorrent::AddTorrentParams params;
+            params.savePath = savepath;
+            params.label = label;
+            const QUrl qurl = QUrl::fromEncoded(url.toUtf8());
```

Comment:
Existing code can also have problems...
And more. No need to create separate commits to fix your PR commits. Just rebase and correct them, otherwise it clogs up qBittorent history.